ui/admin/app/routes/scopes/scope/credential-stores/credential-store/credential-libraries/credential-library.js

```diff
@@ -19,4 +21,27 @@ export default class ScopesScopeCredentialStoresCredentialStoreCredentialLibrari
       reload: true,
     });
   }
+
+  /**
+   * Copies the contents of array fields in order to force the instance
+   * into a dirty state.  This ensures that `model.rollbackAttributes()` reverts
+   * to the original expected array.
+   *
+   * The deep copy implemented here is required to ensure that both the
+   * array itself and its members are all new.
+   *
+   * @param {credentialLibraryModel} credentialLibrary
+   */
+  @action
+  edit(credentialLibrary) {
+    if (credentialLibrary.critical_options) {
+      credentialLibrary.critical_options = copy(
```

Comment:
Non-blocking comment and 100% personal opinion, not a requirement, just a discussion for the sake of learning.

I wonder if the usage of `ember-copy` brings some magic behind the scenes or if in the future we can rely on `array.from()` [docs here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/from) (or similar) to shallow copy arrays and ember-data mark them as dirty state.

Reply:
Unfortunately that only makes a shallow copy, we specifically use the `copy` with the deep copy parameter as `true` for deep copies.